nwbwidgets/spectrum.py

```diff
@@ -0,0 +1,88 @@
+from ndx_spectrum import Spectrum
+import warnings
+import matplotlib.pyplot as plt
+from ipywidgets import widgets
+import numpy as np
+
+
+def show_spectrum(node: Spectrum, **kwargs) -> widgets.Widget:
+    if isinstance(node, Spectrum):
+        data = node.power if 'power' in node.fields else node.phase
+        if len(data.shape)==2:
+            no_channels = data.shape[1]
+        else:
+            no_channels = 1
+        freqs_all = np.asarray(node.frequencies)
+        channel_slider = widgets.IntRangeSlider(
+            min=1,
+            max=no_channels,
+            step=1,
+            description='Channel Range:',
+            disabled=False,
+            continuous_update=False,
+            orientation='horizontal',
+            readout=True,
+            readout_format='d',
+        )
+        freq_slider = widgets.IntRangeSlider(
+            min=0,
+            max=freqs_all[-1],
+            step=1,
+            description='Frequency Range:',
+            disabled=False,
+            continuous_update=False,
+            orientation='horizontal',
+            readout=True,
+            readout_format='d',
+        )
+        out = widgets.Output()
+        with out:
+            widgets.interact(
+                lambda channel_no, frequency_range: get_spectrum_figure(node, channel_no, frequency_range),
+                channel_no=channel_slider,
+                frequency_range=freq_slider)
+        return out
+    else:
+        warnings.warn('neurodatatype not of type: Spectrum')
+
+
+def get_spectrum_figure(spectrum,channel_no, freqs):
```

Comment:
Use docstring to explain what these args should be

Reply:
it takes input from the slider, which can only be a continuous range

Reply:
then wrap it in another function that takes the slider input and changes it to a list of ints

Reply:
line 54 is doing that

Reply:
ok, can you put a docstring in though?